tlslite/tlsconnection.py

```diff
@@ -399,15 +399,23 @@ def _handshakeClientAsyncHelper(self, srpParams, certParams, anonParams,
         #We'll use this for the ClientHello, and if an error occurs
         #parsing the Server Hello, we'll use this version for the response
         self.version = settings.maxVersion
+
+        if settings.useEncryptThenMAC:
+            extensions = [TLSExtension().create(ExtensionType.encrypt_then_mac,
+                                                bytearray(0))]
+        else:
+            extensions = None
```

Comment:
I think it would be cleaner to create the Extensions list within clientSendClientHello(), instead of creating it here and passing it in.